service/src/main/java/com/epam/esm/wrapper/GiftCertificateDtoWrapper.java

```diff
@@ -0,0 +1,47 @@
+package com.epam.esm.wrapper;
+
+
+import com.epam.esm.dto.GiftCertificateDto;
+import com.epam.esm.entities.GiftCertificate;
+import com.epam.esm.entities.Tag;
+import org.springframework.stereotype.Component;
+
+import java.time.ZonedDateTime;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+@Component
+public class GiftCertificateDtoWrapper {
```

Comment:
Why did you decide to use wrapper. In my opinion it is mapper

Reply:
I changed for GiftCertificateDtoMapper